Bug: Flakyness in Aggregate in 0.20.0-rc0

```
--- FAIL: TestLocalMetaWithFilters (0.02s)
    graphql_helper.go:53: Expected the query to succeed, but failed with unprocessable entity: &{couldn't unmarshal json: json: cannot unmarshal number into Go struct field GraphQLError.path of type string
        original result was &graphql.Result{Data:map[string]interface {}{"Aggregate":map[string]interface {}{"Things":map[string]interface {}{"City":[]interface {}{map[string]interface {}{"name":map[string]interface {}{"topOccurrences":interface {}(nil), "type":"string", "count":interface {}(nil)}, "meta":map[string]interface {}{"count":2}, "isCapital":map[string]interface {}{"totalFalse":0, "totalTrue":2, "type":"boolean", "count":2, "percentageFalse":0, "percentageTrue":1}, "population":map[string]interface {}{"mean":2.635e+06, "count":2, "maximum":3.47e+06, "minimum":1.8e+06, "sum":5.27e+06, "type":"int"}, "InCountry":map[string]interface {}{"pointingTo":[]interface {}{"Country"}, "type":"cref"}}}}}}, Errors:[]gqlerrors.FormattedError{gqlerrors.FormattedError{Message:"text: expected property to be of type text, got numerical", Locations:[]location.SourceLocation{location.SourceLocation{Line:34, Column:9}}, Path:[]interface {}{"Aggregate", "Things", "City", 0, "name", "topOccurrences"}, Extensions:map[string]interface {}(nil), originalError:(*errors.errorString)(0xc0005a5060)}}}}
(*test.GraphQLResult)(0xc0004ca950)({
 Result: (map[string]interface {}) (len=1) {
  (string) (len=9) "Aggregate": (map[string]interface {}) (len=1) {
   (string) (len=6) "Things": (map[string]interface {}) (len=1) {
    (string) (len=4) "City": ([]interface {}) (len=1 cap=1) {
     (map[string]interface {}) (len=2) {
      (string) (len=4) "name": (map[string]interface {}) (len=2) {
       (string) (len=5) "count": (interface {}) <nil>,
       (string) (len=10) "__typename": (string) (len=20) "AggregateCitynameObj"
      },
      (string) (len=10) "__typename": (string) (len=13) "AggregateCity"
     }
    }
   }
  }
 }
})
FAIL
```

## Todos

* [x] address issue if only meta count is specified
* [x] address flakyness leading to error message above

--- a/adapters/repos/db/aggregations_fixtures_for_test.go
+++ b/adapters/repos/db/aggregations_fixtures_for_test.go
@@ -9,6 +9,7 @@
 //  CONTACT: hello@semi.technology
 //
 
+//go:build integrationTest
 // +build integrationTest
 
 package db
@@ -62,6 +63,22 @@ var companyClass = &models.Class{
 		{
 			Name:     "makesProduct",
 			DataType: []string{"AggregationsTestProduct"},
+		},
+	},
+}
+
+var arrayTypesClass = &models.Class{
+	Class:               "AggregationsTestArrayTypes",
+	VectorIndexConfig:   hnsw.NewDefaultUserConfig(),
+	InvertedIndexConfig: invertedConfig(),
+	Properties: []*models.Property{
+		{
+			Name:     "strings",
+			DataType: []string{"string[]"},
+		},
+		{
+			Name:     "numbers",
+			DataType: []string{"number[]"},
 		},
 	},
 }
@@ -144,5 +161,16 @@ var companies = []map[string]interface{}{
 		"dividendYield": 1.1,
 		"price":         int64(70),
 		"listedInIndex": true,
+	},
+}
+
+var arrayTypes = []map[string]interface{}{
+	{
+		"strings": []string{"a", "b", "c"},
+		"numbers": []float64{1.0, 2.0, 3.0},
+	},
+	{
+		"strings": []string{"a"},
+		"numbers": []float64{1.0, 2.0},
 	},
 }

--- a/adapters/repos/db/aggregations_integration_test.go
+++ b/adapters/repos/db/aggregations_integration_test.go
@@ -9,6 +9,7 @@
 //  CONTACT: hello@semi.technology
 //
 
+//go:build integrationTest
 // +build integrationTest
 
 package db
@@ -76,6 +77,7 @@ func prepareCompanyTestSchemaAndData(repo *DB,
 				Classes: []*models.Class{
 					productClass,
 					companyClass,
+					arrayTypesClass,
 				},
 			},
 		}
@@ -85,6 +87,8 @@ func prepareCompanyTestSchemaAndData(repo *DB,
 				migrator.AddClass(context.Background(), productClass))
 			require.Nil(t,
 				migrator.AddClass(context.Background(), companyClass))
+			require.Nil(t,
+				migrator.AddClass(context.Background(), arrayTypesClass))
 		})
 
 		schemaGetter.schema = schema
@@ -108,6 +112,20 @@ func prepareCompanyTestSchemaAndData(repo *DB,
 				t.Run(fmt.Sprintf("importing company %d", i), func(t *testing.T) {
 					fixture := models.Object{
 						Class:      companyClass.Class,
+						ID:         strfmt.UUID(uuid.Must(uuid.NewRandom()).String()),
+						Properties: schema,
+					}
+					require.Nil(t,
+						repo.PutObject(context.Background(), &fixture, []float32{0.1, 0.1, 0.1, 0.1}))
+				})
+			}
+		})
+
+		t.Run("import array types", func(t *testing.T) {
+			for i, schema := range arrayTypes {
+				t.Run(fmt.Sprintf("importing array type %d", i), func(t *testing.T) {
+					fixture := models.Object{
+						Class:      arrayTypesClass.Class,
 						ID:         strfmt.UUID(uuid.Must(uuid.NewRandom()).String()),
 						Properties: schema,
 					}
@@ -843,6 +861,96 @@ func testNumericalAggregationsWithGrouping(repo *DB) func(t *testing.T) {
 			assert.InDeltaMapValues(t, expectedProps["price"].NumericalAggregations,
 				actualProps["price"].NumericalAggregations, 0.001)
 		})
+
+		t.Run("array types, single aggregator strings", func(t *testing.T) {
+			params := traverser.AggregateParams{
+				ClassName: schema.ClassName(arrayTypesClass.Class),
+				GroupBy: &filters.Path{
+					Class:    schema.ClassName(arrayTypesClass.Class),
+					Property: schema.PropertyName("strings"),
+				},
+				IncludeMetaCount: true,
+			}
+
+			res, err := repo.Aggregate(context.Background(), params)
+			require.Nil(t, err)
+
+			expectedResult := &aggregation.Result{
+				Groups: []aggregation.Group{
+					aggregation.Group{
+						Count: 2,
+						GroupedBy: &aggregation.GroupedBy{
+							Path:  []string{"strings"},
+							Value: "a",
+						},
+						Properties: map[string]aggregation.Property{},
+					},
+					aggregation.Group{
+						Count: 1,
+						GroupedBy: &aggregation.GroupedBy{
+							Path:  []string{"strings"},
+							Value: "b",
+						},
+						Properties: map[string]aggregation.Property{},
+					},
+					aggregation.Group{
+						Count: 1,
+						GroupedBy: &aggregation.GroupedBy{
+							Path:  []string{"strings"},
+							Value: "c",
+						},
+						Properties: map[string]aggregation.Property{},
+					},
+				},
+			}
+
+			assert.ElementsMatch(t, expectedResult.Groups, res.Groups)
+		})
+
+		t.Run("array types, single aggregator numbers", func(t *testing.T) {
+			params := traverser.AggregateParams{
+				ClassName: schema.ClassName(arrayTypesClass.Class),
+				GroupBy: &filters.Path{
+					Class:    schema.ClassName(arrayTypesClass.Class),
+					Property: schema.PropertyName("numbers"),
+				},
+				IncludeMetaCount: true,
+			}
+
+			res, err := repo.Aggregate(context.Background(), params)
+			require.Nil(t, err)
+
+			expectedResult := &aggregation.Result{
+				Groups: []aggregation.Group{
+					aggregation.Group{
+						Count: 2,
+						GroupedBy: &aggregation.GroupedBy{
+							Path:  []string{"numbers"},
+							Value: float64(1.0),
+						},
+						Properties: map[string]aggregation.Property{},
+					},
+					aggregation.Group{
+						Count: 2,
+						GroupedBy: &aggregation.GroupedBy{
+							Path:  []string{"numbers"},
+							Value: float64(2.0),
+						},
+						Properties: map[string]aggregation.Property{},
+					},
+					aggregation.Group{
+						Count: 1,
+						GroupedBy: &aggregation.GroupedBy{
+							Path:  []string{"numbers"},
+							Value: float64(3.0),
+						},
+						Properties: map[string]aggregation.Property{},
+					},
+				},
+			}
+
+			assert.ElementsMatch(t, expectedResult.Groups, res.Groups)
+		})
 	}
 }
 
@@ -1377,6 +1485,122 @@ func testNumericalAggregationsWithoutGrouping(repo *DB) func(t *testing.T) {
 										aggregation.TextOccurrence{
 											Value:  "Food",
 											Occurs: 1,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			assert.Equal(t, expectedResult.Groups, res.Groups)
+		})
+
+		t.Run("array types, only meta count, no other aggregations", func(t *testing.T) {
+			params := traverser.AggregateParams{
+				ClassName:        schema.ClassName(arrayTypesClass.Class),
+				IncludeMetaCount: true,
+				GroupBy:          nil, // explicitly set to nil
+			}
+
+			res, err := repo.Aggregate(context.Background(), params)
+			require.Nil(t, err)
+
+			expectedResult := &aggregation.Result{
+				Groups: []aggregation.Group{
+					aggregation.Group{
+						GroupedBy: nil,
+						Count:     2,
+					},
+				},
+			}
+
+			require.NotNil(t, res)
+			assert.Equal(t, expectedResult.Groups, res.Groups)
+		})
+
+		t.Run("array types, single aggregator numbers", func(t *testing.T) {
+			params := traverser.AggregateParams{
+				ClassName: schema.ClassName(arrayTypesClass.Class),
+				GroupBy:   nil, // explicitly set to nil
+				Properties: []traverser.AggregateProperty{
+					traverser.AggregateProperty{
+						Name: schema.PropertyName("numbers"),
+						Aggregators: []traverser.Aggregator{
+							traverser.MeanAggregator,
+							traverser.MaximumAggregator,
+							traverser.MinimumAggregator,
+							traverser.SumAggregator,
+							traverser.ModeAggregator,
+							traverser.MedianAggregator,
+							traverser.CountAggregator,
+							traverser.TypeAggregator, // ignored in the repo, but can't block
+						},
+					},
+				},
+			}
+
+			res, err := repo.Aggregate(context.Background(), params)
+			require.Nil(t, err)
+
+			expectedResult := &aggregation.Result{
+				Groups: []aggregation.Group{
+					aggregation.Group{
+						GroupedBy: nil,
+						Properties: map[string]aggregation.Property{
+							"numbers": aggregation.Property{
+								Type: aggregation.PropertyTypeNumerical,
+								NumericalAggregations: map[string]float64{
+									"mean":    1.8,
+									"maximum": 3.0,
+									"minimum": 1.0,
+									"sum":     9.0,
+									"mode":    1.0,
+									"median":  3.0,
+									"count":   5,
+								},
+							},
+						},
+					},
+				},
+			}
+
+			assert.Equal(t, expectedResult.Groups, res.Groups)
+		})
+
+		t.Run("array types, single aggregator strings", func(t *testing.T) {
+			params := traverser.AggregateParams{
+				ClassName: schema.ClassName(arrayTypesClass.Class),
+				GroupBy:   nil, // explicitly set to nil
+				Properties: []traverser.AggregateProperty{
+					traverser.AggregateProperty{
+						Name: schema.PropertyName("strings"),
+						Aggregators: []traverser.Aggregator{
+							// limit is very restrictive
+							traverser.NewTopOccurrencesAggregator(ptInt(1)),
+							traverser.TypeAggregator, // ignored in the repo, but can't block
+						},
+					},
+				},
+			}
+
+			res, err := repo.Aggregate(context.Background(), params)
+			require.Nil(t, err)
+
+			expectedResult := &aggregation.Result{
+				Groups: []aggregation.Group{
+					aggregation.Group{
+						GroupedBy: nil,
+						Properties: map[string]aggregation.Property{
+							"strings": aggregation.Property{
+								Type: aggregation.PropertyTypeText,
+								TextAggregation: aggregation.Text{
+									Count: 4,
+									Items: []aggregation.TextOccurrence{
+										aggregation.TextOccurrence{
+											Value:  "a",
+											Occurs: 2,
 										},
 									},
 								},

--- a/adapters/repos/db/aggregator/grouper.go
+++ b/adapters/repos/db/aggregator/grouper.go
@@ -98,10 +98,26 @@ func (g *grouper) addElement(obj *storobj.Object) error {
 		return nil
 	}
 
-	ids := g.values[item]
-	ids = append(ids, obj.DocID())
-	g.values[item] = ids
+	switch val := item.(type) {
+	case []string:
+		for i := range val {
+			g.addItem(val[i], obj.DocID())
+		}
+	case []float64:
+		for i := range val {
+			g.addItem(val[i], obj.DocID())
+		}
+	default:
+		g.addItem(val, obj.DocID())
+	}
+
 	return nil
+}
+
+func (g *grouper) addItem(item interface{}, docID uint64) {
+	ids := g.values[item]
+	ids = append(ids, docID)
+	g.values[item] = ids
 }
 
 func (g *grouper) aggregateAndSelect() ([]group, error) {

--- a/test/acceptance/graphql_resolvers/aggregate_test.go
+++ b/test/acceptance/graphql_resolvers/aggregate_test.go
@@ -11,287 +11,370 @@
 
 package test
 
-// TODO: Fix flakyness https://github.com/semi-technologies/weaviate/issues/978
+import (
+	"encoding/json"
+	"testing"
 
-// import (
-// 	"encoding/json"
-// 	"testing"
+	"github.com/semi-technologies/weaviate/test/acceptance/helper"
+	"github.com/stretchr/testify/assert"
+)
 
-// 	"github.com/semi-technologies/weaviate/test/acceptance/helper"
-// 	"github.com/stretchr/testify/assert"
-// )
+func aggregatesWithoutGroupingOrFilters(t *testing.T) {
+	result := AssertGraphQL(t, helper.RootAuth, `
+		{
+			Aggregate{
+				City {
+					meta {
+						count
+					}
+					isCapital {
+						count
+						percentageFalse
+						percentageTrue
+						totalFalse
+						totalTrue
+						type
+					}
+					population {
+						mean
+						count
+						maximum
+						minimum
+						sum
+						type
+					}
+					inCountry {
+						pointingTo
+						type
+					}
+					name {
+						topOccurrences {
+							occurs
+							value
+						}
+						type
+						count
+					}
+				}
+			}
+		}
+	`)
 
-// func Test_Aggregates_WithoutGroupingOrFilters(t *testing.T) {
+	t.Run("meta count", func(t *testing.T) {
+		meta := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["meta"]
+		count := meta.(map[string]interface{})["count"]
+		expected := json.Number("5")
+		assert.Equal(t, expected, count)
+	})
 
-// 	result := AssertGraphQL(t, helper.RootAuth, `
-// 		{
-// 			Aggregate{
-// 				Things {
-// 					City {
-// 						meta {
-// 							count
-// 						}
-// 						isCapital {
-// 							count
-// 							percentageFalse
-// 							percentageTrue
-// 							totalFalse
-// 							totalTrue
-// 							type
-// 						}
-// 						population {
-// 							mean
-// 							count
-// 							maximum
-// 							minimum
-// 							sum
-// 							type
-// 						}
-// 						InCountry {
-// 							pointingTo
-// 							type
-// 						}
-// 						name {
-// 							topOccurrences {
-// 								occurs
-// 								value
-// 							}
-// 							type
-// 							count
-// 						}
-// 					}
-// 				}
-// 			}
-// 		}
-// 	`)
+	t.Run("boolean props", func(t *testing.T) {
+		isCapital := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["isCapital"]
+		expected := map[string]interface{}{
+			"count":           json.Number("5"),
+			"percentageTrue":  json.Number("0.4"),
+			"percentageFalse": json.Number("0.6"),
+			"totalTrue":       json.Number("2"),
+			"totalFalse":      json.Number("3"),
+			"type":            "boolean",
+		}
+		assert.Equal(t, expected, isCapital)
+	})
 
-// 	t.Run("meta count", func(t *testing.T) {
-// 		meta := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["meta"]
-// 		count := meta.(map[string]interface{})["count"]
-// 		expected := json.Number("4")
-// 		assert.Equal(t, expected, count)
-// 	})
+	t.Run("int/number props", func(t *testing.T) {
+		isCapital := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["population"]
+		expected := map[string]interface{}{
+			"mean":    json.Number("1294000"),
+			"count":   json.Number("5"),
+			"maximum": json.Number("3470000"),
+			"minimum": json.Number("0"),
+			"sum":     json.Number("6470000"),
+			"type":    "int",
+		}
+		assert.Equal(t, expected, isCapital)
+	})
 
-// 	t.Run("boolean props", func(t *testing.T) {
-// 		isCapital := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["isCapital"]
-// 		expected := map[string]interface{}{
-// 			"count":           json.Number("4"),
-// 			"percentageTrue":  json.Number("0.5"),
-// 			"percentageFalse": json.Number("0.5"),
-// 			"totalTrue":       json.Number("2"),
-// 			"totalFalse":      json.Number("2"),
-// 			"type":            "boolean",
-// 		}
-// 		assert.Equal(t, expected, isCapital)
-// 	})
+	t.Run("ref prop", func(t *testing.T) {
+		inCountry := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["inCountry"]
+		expected := map[string]interface{}{
+			"pointingTo": []interface{}{"Country"},
+			"type":       "cref",
+		}
+		assert.Equal(t, expected, inCountry)
+	})
 
-// 	t.Run("int/number props", func(t *testing.T) {
-// 		isCapital := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["population"]
-// 		expected := map[string]interface{}{
-// 			"mean":    json.Number("1917500"),
-// 			"count":   json.Number("4"),
-// 			"maximum": json.Number("3470000"),
-// 			"minimum": json.Number("600000"),
-// 			"sum":     json.Number("7670000"),
-// 			"type":    "int",
-// 		}
-// 		assert.Equal(t, expected, isCapital)
-// 	})
+	t.Run("string prop", func(t *testing.T) {
+		name := result.Get("Aggregate", "City").
+			AsSlice()[0].(map[string]interface{})["name"].(map[string]interface{})
+		typeField := name["type"]
+		topOccurrences := name["topOccurrences"]
 
-// 	t.Run("ref prop", func(t *testing.T) {
-// 		inCountry := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["InCountry"]
-// 		expected := map[string]interface{}{
-// 			"pointingTo": []interface{}{"Country"},
-// 			"type":       "cref",
-// 		}
-// 		assert.Equal(t, expected, inCountry)
-// 	})
+		assert.Equal(t, "string", typeField)
 
-// 	t.Run("string prop", func(t *testing.T) {
-// 		name := result.Get("Aggregate", "Things", "City").
-// 			AsSlice()[0].(map[string]interface{})["name"].(map[string]interface{})
-// 		typeField := name["type"]
-// 		// count := name["count"]
-// 		topOccurrences := name["topOccurrences"]
+		expectedTopOccurrences := []interface{}{
+			map[string]interface{}{
+				"value":  "Amsterdam",
+				"occurs": json.Number("1"),
+			},
+			map[string]interface{}{
+				"value":  "Dusseldorf",
+				"occurs": json.Number("1"),
+			},
+			map[string]interface{}{
+				"value":  "Rotterdam",
+				"occurs": json.Number("1"),
+			},
+			map[string]interface{}{
+				"value":  "Berlin",
+				"occurs": json.Number("1"),
+			},
+			map[string]interface{}{
+				"value":  "Null Island",
+				"occurs": json.Number("1"),
+			},
+		}
+		assert.ElementsMatch(t, expectedTopOccurrences, topOccurrences)
+	})
+}
 
-// 		// TODO: fix string count
-// 		// assert.Equal(t, json.Number("4"), count)
-// 		assert.Equal(t, "string", typeField)
+func localMetaWithFilters(t *testing.T) {
+	result := AssertGraphQL(t, helper.RootAuth, `
+		{
+			Aggregate{
+				City (where: {
+					valueBoolean: true,
+					operator: Equal,
+					path: ["isCapital"]
+				}){
+					meta {
+						count
+					}
+					isCapital {
+						count
+						percentageFalse
+						percentageTrue
+						totalFalse
+						totalTrue
+						type
+					}
+					population {
+						mean
+						count
+						maximum
+						minimum
+						sum
+						type
+					}
+					inCountry {
+						pointingTo
+						type
+					}
+					name {
+						topOccurrences {
+							occurs
+							value
+						}
+						type
+						count
+					}
+				}
+			}
+		}
+	`)
 
-// 		expectedTopOccurrences := []interface{}{
-// 			map[string]interface{}{
-// 				"value":  "Amsterdam",
-// 				"occurs": json.Number("1"),
-// 			},
-// 			map[string]interface{}{
-// 				"value":  "Dusseldorf",
-// 				"occurs": json.Number("1"),
-// 			},
-// 			map[string]interface{}{
-// 				"value":  "Rotterdam",
-// 				"occurs": json.Number("1"),
-// 			},
-// 			map[string]interface{}{
-// 				"value":  "Berlin",
-// 				"occurs": json.Number("1"),
-// 			},
-// 		}
-// 		assert.ElementsMatch(t, expectedTopOccurrences, topOccurrences)
-// 	})
-// }
+	t.Run("meta count", func(t *testing.T) {
+		meta := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["meta"]
+		count := meta.(map[string]interface{})["count"]
+		expected := json.Number("2")
+		assert.Equal(t, expected, count)
+	})
 
-// func TestLocalMetaWithFilters(t *testing.T) {
-// 	result := AssertGraphQL(t, helper.RootAuth, `
-// 		{
-// 				Aggregate{
-// 					Things {
-// 						City (where: {
-// 							valueBoolean: true,
-// 							operator: Equal,
-// 							path: ["isCapital"]
-// 						}){
-// 							meta {
-// 								count
-// 							}
-// 							isCapital {
-// 								count
-// 								percentageFalse
-// 								percentageTrue
-// 								totalFalse
-// 								totalTrue
-// 								type
-// 							}
-// 							population {
-// 								mean
-// 								count
-// 								maximum
-// 								minimum
-// 								sum
-// 								type
-// 							}
-// 							InCountry {
-// 								pointingTo
-// 								type
-// 							}
-// 							name {
-// 								topOccurrences {
-// 									occurs
-// 									value
-// 								}
-// 								type
-// 								count
-// 							}
-// 						}
-// 					}
-// 				}
-// 			}
-// 	`)
+	t.Run("boolean props", func(t *testing.T) {
+		isCapital := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["isCapital"]
+		expected := map[string]interface{}{
+			"count":           json.Number("2"),
+			"percentageTrue":  json.Number("1"),
+			"percentageFalse": json.Number("0"),
+			"totalTrue":       json.Number("2"),
+			"totalFalse":      json.Number("0"),
+			"type":            "boolean",
+		}
+		assert.Equal(t, expected, isCapital)
+	})
 
-// 	t.Run("meta count", func(t *testing.T) {
-// 		meta := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["meta"]
-// 		count := meta.(map[string]interface{})["count"]
-// 		expected := json.Number("2")
-// 		assert.Equal(t, expected, count)
-// 	})
+	t.Run("int/number props", func(t *testing.T) {
+		population := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["population"]
+		expected := map[string]interface{}{
+			"mean":    json.Number("2635000"),
+			"count":   json.Number("2"),
+			"maximum": json.Number("3470000"),
+			"minimum": json.Number("1800000"),
+			"sum":     json.Number("5270000"),
+			"type":    "int",
+		}
+		assert.Equal(t, expected, population)
+	})
 
-// 	t.Run("boolean props", func(t *testing.T) {
-// 		isCapital := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["isCapital"]
-// 		expected := map[string]interface{}{
-// 			"count":           json.Number("2"),
-// 			"percentageTrue":  json.Number("1"),
-// 			"percentageFalse": json.Number("0"),
-// 			"totalTrue":       json.Number("2"),
-// 			"totalFalse":      json.Number("0"),
-// 			"type":            "boolean",
-// 		}
-// 		assert.Equal(t, expected, isCapital)
-// 	})
+	t.Run("ref prop", func(t *testing.T) {
+		inCountry := result.Get("Aggregate", "City").AsSlice()[0].(map[string]interface{})["inCountry"]
+		expected := map[string]interface{}{
+			"pointingTo": []interface{}{"Country"},
+			"type":       "cref",
+		}
+		assert.Equal(t, expected, inCountry)
+	})
 
-// 	t.Run("int/number props", func(t *testing.T) {
-// 		population := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["population"]
-// 		expected := map[string]interface{}{
-// 			"mean":    json.Number("2635000"),
-// 			"count":   json.Number("2"),
-// 			"maximum": json.Number("3470000"),
-// 			"minimum": json.Number("1800000"),
-// 			"sum":     json.Number("5270000"),
-// 			"type":    "int",
-// 		}
-// 		assert.Equal(t, expected, population)
-// 	})
+	t.Run("string prop", func(t *testing.T) {
+		name := result.Get("Aggregate", "City").
+			AsSlice()[0].(map[string]interface{})["name"].(map[string]interface{})
+		typeField := name["type"]
+		topOccurrences := name["topOccurrences"]
 
-// 	t.Run("ref prop", func(t *testing.T) {
-// 		inCountry := result.Get("Aggregate", "Things", "City").AsSlice()[0].(map[string]interface{})["InCountry"]
-// 		expected := map[string]interface{}{
-// 			"pointingTo": []interface{}{"Country"},
-// 			"type":       "cref",
-// 		}
-// 		assert.Equal(t, expected, inCountry)
-// 	})
+		assert.Equal(t, "string", typeField)
 
-// 	t.Run("string prop", func(t *testing.T) {
-// 		name := result.Get("Aggregate", "Things", "City").
-// 			AsSlice()[0].(map[string]interface{})["name"].(map[string]interface{})
-// 		typeField := name["type"]
-// 		// count := name["count"]
-// 		topOccurrences := name["topOccurrences"]
+		expectedTopOccurrences := []interface{}{
+			map[string]interface{}{
+				"value":  "Amsterdam",
+				"occurs": json.Number("1"),
+			},
+			map[string]interface{}{
+				"value":  "Berlin",
+				"occurs": json.Number("1"),
+			},
+		}
+		assert.ElementsMatch(t, expectedTopOccurrences, topOccurrences)
+	})
+}
 
-// 		// TODO: fix string count
-// 		// assert.Equal(t, json.Number("2"), count)
-// 		assert.Equal(t, "string", typeField)
+// This test prevents a regression on the fix for
+// https://github.com/semi-technologies/weaviate/issues/824
+func localMeta_StringPropsNotSetEverywhere(t *testing.T) {
+	AssertGraphQL(t, helper.RootAuth, `
+		{
+			Aggregate {
+				City {
+					name {
+						topOccurrences {
+							occurs
+							value
+						}
+					}
+				}
+			}
+		}
+	`)
+}
 
-// 		expectedTopOccurrences := []interface{}{
-// 			map[string]interface{}{
-// 				"value":  "Amsterdam",
-// 				"occurs": json.Number("1"),
-// 			},
-// 			map[string]interface{}{
-// 				"value":  "Berlin",
-// 				"occurs": json.Number("1"),
-// 			},
-// 		}
-// 		assert.ElementsMatch(t, expectedTopOccurrences, topOccurrences)
-// 	})
-// }
+func aggregatesArrayClassWithoutGroupingOrFilters(t *testing.T) {
+	result := AssertGraphQL(t, helper.RootAuth, `
+		{
+			Aggregate{
+				ArrayClass {
+					meta {
+						count
+					}
+					numbers {
+						mean
+						count
+						maximum
+						minimum
+						sum
+						type
+					}
+					strings {
+						topOccurrences {
+							occurs
+							value
+						}
+						type
+						count
+					}
+				}
+			}
+		}
+	`)
 
-// // This test prevents a regression on the fix for
-// // https://github.com/semi-technologies/weaviate/issues/824
-// func TestLocalMeta_StringPropsNotSetEverywhere(t *testing.T) {
-// 	AssertGraphQL(t, helper.RootAuth, `
-// 		{
-// 				Aggregate {
-// 					Actions {
-// 						Event {
-// 							name {
-// 								topOccurrences {
-// 									occurs
-// 									value
-// 								}
-// 							}
-// 						}
-// 					}
-// 				}
-// 		}
-// 	`)
-// }
+	t.Run("meta count", func(t *testing.T) {
+		meta := result.Get("Aggregate", "ArrayClass").AsSlice()[0].(map[string]interface{})["meta"]
+		count := meta.(map[string]interface{})["count"]
+		expected := json.Number("3")
+		assert.Equal(t, expected, count)
+	})
 
-// // // This test prevents a regression on the fix for
-// // // https://github.com/semi-technologies/weaviate/issues/824
-// // func TestLocalMeta_TextPropsNotSetEverywhere(t *testing.T) {
-// // 	AssertGraphQL(t, helper.RootAuth, `
-// // 		{
-// // 				Aggregate {
-// // 					Actions {
-// // 						Event {
-// // 							description {
-// // 								topOccurrences {
-// // 									occurs
-// // 									value
-// // 								}
-// // 							}
-// // 						}
-// // 					}
-// // 				}
-// // 		}
-// // 	`)
-// // }
+	t.Run("int[]/number[] props", func(t *testing.T) {
+		isCapital := result.Get("Aggregate", "ArrayClass").AsSlice()[0].(map[string]interface{})["numbers"]
+		expected := map[string]interface{}{
+			"mean":    json.Number("1.6666666666666667"),
+			"count":   json.Number("6"),
+			"maximum": json.Number("3"),
+			"minimum": json.Number("1"),
+			"sum":     json.Number("10"),
+			"type":    "number[]",
+		}
+		assert.Equal(t, expected, isCapital)
+	})
+
+	t.Run("string[]/text[] prop", func(t *testing.T) {
+		name := result.Get("Aggregate", "ArrayClass").
+			AsSlice()[0].(map[string]interface{})["strings"].(map[string]interface{})
+		typeField := name["type"]
+		topOccurrences := name["topOccurrences"]
+
+		assert.Equal(t, "string[]", typeField)
+
+		expectedTopOccurrences := []interface{}{
+			map[string]interface{}{
+				"value":  "a",
+				"occurs": json.Number("3"),
+			},
+			map[string]interface{}{
+				"value":  "b",
+				"occurs": json.Number("2"),
+			},
+			map[string]interface{}{
+				"value":  "c",
+				"occurs": json.Number("1"),
+			},
+		}
+		assert.ElementsMatch(t, expectedTopOccurrences, topOccurrences)
+	})
+}
+
+func aggregatesArrayClassWithGrouping(t *testing.T) {
+	result := AssertGraphQL(t, helper.RootAuth, `
+	{
+		Aggregate{
+			ArrayClass(groupBy:["numbers"]){
+				meta{
+					count
+				}
+				groupedBy{
+					value
+				}
+			}
+		}
+	}
+	`)
+
+	t.Run("groupedBy result", func(t *testing.T) {
+		groupedByResults := result.Get("Aggregate", "ArrayClass").AsSlice()
+		assert.Equal(t, 3, len(groupedByResults))
+
+		for _, res := range groupedByResults {
+			meta := res.(map[string]interface{})["meta"]
+			count := meta.(map[string]interface{})["count"]
+			groupedBy := res.(map[string]interface{})["groupedBy"]
+			value := groupedBy.(map[string]interface{})["value"]
+			valueString := value.(string)
+
+			if valueString == "1" {
+				assert.Equal(t, json.Number("3"), count)
+			}
+			if valueString == "2" {
+				assert.Equal(t, json.Number("2"), count)
+			}
+			if valueString == "3" {
+				assert.Equal(t, json.Number("1"), count)
+			}
+		}
+	})
+}

--- a/test/acceptance/graphql_resolvers/setup_test.go
+++ b/test/acceptance/graphql_resolvers/setup_test.go
@@ -31,6 +31,7 @@ func Test_GraphQL(t *testing.T) {
 	t.Run("import test data (companies)", addTestDataCompanies)
 	t.Run("import test data (person)", addTestDataPersons)
 	t.Run("import test data (custom vector class)", addTestDataCVC)
+	t.Run("import test data (array class)", addTestDataArrayClasses)
 
 	// tests
 	t.Run("getting objects", gettingObjects)
@@ -38,6 +39,11 @@ func Test_GraphQL(t *testing.T) {
 	t.Run("getting objects with geo filters", gettingObjectsWithGeoFilters)
 	t.Run("getting objects with grouping", gettingObjectsWithGrouping)
 	t.Run("getting objects with additional props", gettingObjectsWithAdditionalProps)
+	t.Run("aggregates without grouping or filters", aggregatesWithoutGroupingOrFilters)
+	t.Run("aggregates local meta with filters", localMetaWithFilters)
+	t.Run("aggregates local meta string props not set everywhere", localMeta_StringPropsNotSetEverywhere)
+	t.Run("aggregates array class without grouping or filters", aggregatesArrayClassWithoutGroupingOrFilters)
+	t.Run("aggregates array class with grouping", aggregatesArrayClassWithGrouping)
 
 	// tear down
 	deleteObjectClass(t, "Person")
@@ -45,6 +51,7 @@ func Test_GraphQL(t *testing.T) {
 	deleteObjectClass(t, "City")
 	deleteObjectClass(t, "Airport")
 	deleteObjectClass(t, "Company")
+	deleteObjectClass(t, "ArrayClass")
 
 	// only run after everything else is deleted, this way, we can also run an
 	// all-class Explore since all vectors which are now left have the same
@@ -92,7 +99,7 @@ func addTestSchema(t *testing.T) {
 			},
 		},
 		Properties: []*models.Property{
-			&models.Property{
+			{
 				Name:     "name",
 				DataType: []string{"string"},
 			},
@@ -107,21 +114,25 @@ func addTestSchema(t *testing.T) {
 			},
 		},
 		Properties: []*models.Property{
-			&models.Property{
+			{
 				Name:     "name",
 				DataType: []string{"string"},
 			},
-			&models.Property{
+			{
 				Name:     "inCountry",
 				DataType: []string{"Country"},
 			},
-			&models.Property{
+			{
 				Name:     "population",
 				DataType: []string{"int"},
 			},
-			&models.Property{
+			{
 				Name:     "location",
 				DataType: []string{"geoCoordinates"},
+			},
+			{
+				Name:     "isCapital",
+				DataType: []string{"boolean"},
 			},
 		},
 	})
@@ -134,15 +145,15 @@ func addTestSchema(t *testing.T) {
 			},
 		},
 		Properties: []*models.Property{
-			&models.Property{
+			{
 				Name:     "code",
 				DataType: []string{"string"},
 			},
-			&models.Property{
+			{
 				Name:     "phone",
 				DataType: []string{"phoneNumber"},
 			},
-			&models.Property{
+			{
 				Name:     "inCity",
 				DataType: []string{"City"},
 			},
@@ -157,7 +168,7 @@ func addTestSchema(t *testing.T) {
 			},
 		},
 		Properties: []*models.Property{
-			&models.Property{
+			{
 				Name:     "name",
 				DataType: []string{"string"},
 				ModuleConfig: map[string]interface{}{
@@ -166,7 +177,7 @@ func addTestSchema(t *testing.T) {
 					},
 				},
 			},
-			&models.Property{
+			{
 				Name:     "inCity",
 				DataType: []string{"City"},
 				ModuleConfig: map[string]interface{}{
@@ -186,7 +197,7 @@ func addTestSchema(t *testing.T) {
 			},
 		},
 		Properties: []*models.Property{
-			&models.Property{
+			{
 				Name:     "name",
 				DataType: []string{"string"},
 				ModuleConfig: map[string]interface{}{
@@ -195,7 +206,7 @@ func addTestSchema(t *testing.T) {
 					},
 				},
 			},
-			&models.Property{
+			{
 				Name:     "livesIn",
 				DataType: []string{"City"},
 				ModuleConfig: map[string]interface{}{
@@ -211,6 +222,25 @@ func addTestSchema(t *testing.T) {
 		Class:      "CustomVectorClass",
 		Vectorizer: "none",
 		Properties: []*models.Property{},
+	})
+
+	createObjectClass(t, &models.Class{
+		Class: "ArrayClass",
+		ModuleConfig: map[string]interface{}{
+			"text2vec-contextionary": map[string]interface{}{
+				"vectorizeClassName": true,
+			},
+		},
+		Properties: []*models.Property{
+			{
+				Name:     "strings",
+				DataType: []string{"string[]"},
+			},
+			{
+				Name:     "numbers",
+				DataType: []string{"number[]"},
+			},
+		},
 	})
 }
 
@@ -264,6 +294,7 @@ func addTestDataCityAirport(t *testing.T) {
 					"beacon": crossref.New("localhost", netherlands).String(),
 				},
 			},
+			"isCapital": true,
 		},
 	})
 	createObject(t, &models.Object{
@@ -277,6 +308,7 @@ func addTestDataCityAirport(t *testing.T) {
 					"beacon": crossref.New("localhost", netherlands).String(),
 				},
 			},
+			"isCapital": false,
 		},
 	})
 	createObject(t, &models.Object{
@@ -290,6 +322,7 @@ func addTestDataCityAirport(t *testing.T) {
 					"beacon": crossref.New("localhost", germany).String(),
 				},
 			},
+			"isCapital": true,
 		},
 	})
 	createObject(t, &models.Object{
@@ -307,6 +340,7 @@ func addTestDataCityAirport(t *testing.T) {
 				"latitude":  51.225556,
 				"longitude": 6.782778,
 			},
+			"isCapital": false,
 		},
 	})
 
@@ -320,6 +354,7 @@ func addTestDataCityAirport(t *testing.T) {
 				"latitude":  0,
 				"longitude": 0,
 			},
+			"isCapital": false,
 		},
 	})
 
@@ -406,15 +441,15 @@ func addTestDataCompanies(t *testing.T) {
 	}
 
 	companies := []companyTemplate{
-		companyTemplate{id: microsoft1, name: "Microsoft Inc.", inCity: []strfmt.UUID{dusseldorf}},
-		companyTemplate{id: microsoft2, name: "Microsoft Incorporated", inCity: []strfmt.UUID{dusseldorf, amsterdam}},
-		companyTemplate{id: microsoft3, name: "Microsoft", inCity: []strfmt.UUID{berlin}},
-		companyTemplate{id: apple1, name: "Apple Inc."},
-		companyTemplate{id: apple2, name: "Apple Incorporated"},
-		companyTemplate{id: apple3, name: "Apple"},
-		companyTemplate{id: google1, name: "Google Inc."},
-		companyTemplate{id: google2, name: "Google Incorporated"},
-		companyTemplate{id: google3, name: "Google"},
+		{id: microsoft1, name: "Microsoft Inc.", inCity: []strfmt.UUID{dusseldorf}},
+		{id: microsoft2, name: "Microsoft Incorporated", inCity: []strfmt.UUID{dusseldorf, amsterdam}},
+		{id: microsoft3, name: "Microsoft", inCity: []strfmt.UUID{berlin}},
+		{id: apple1, name: "Apple Inc."},
+		{id: apple2, name: "Apple Incorporated"},
+		{id: apple3, name: "Apple"},
+		{id: google1, name: "Google Inc."},
+		{id: google2, name: "Google Incorporated"},
+		{id: google3, name: "Google"},
 	}
 
 	// companies
@@ -507,4 +542,41 @@ func addTestDataCVC(t *testing.T) {
 		},
 	})
 	assertGetObjectEventually(t, cvc3)
+}
+
+func addTestDataArrayClasses(t *testing.T) {
+	var (
+		arrayClassID1 strfmt.UUID = "cfa3b21e-ca5f-4db7-a412-5fc6a23c534a"
+		arrayClassID2 strfmt.UUID = "cfa3b21e-ca5f-4db7-a412-5fc6a23c534b"
+		arrayClassID3 strfmt.UUID = "cfa3b21e-ca5f-4db7-a412-5fc6a23c534c"
+	)
+	createObject(t, &models.Object{
+		Class: "ArrayClass",
+		ID:    arrayClassID1,
+		Properties: map[string]interface{}{
+			"strings": []string{"a", "b", "c"},
+			"numbers": []float64{1.0, 2.0, 3.0},
+		},
+	})
+	assertGetObjectEventually(t, arrayClassID1)
+
+	createObject(t, &models.Object{
+		Class: "ArrayClass",
+		ID:    arrayClassID2,
+		Properties: map[string]interface{}{
+			"strings": []string{"a", "b"},
+			"numbers": []float64{1.0, 2.0},
+		},
+	})
+	assertGetObjectEventually(t, arrayClassID2)
+
+	createObject(t, &models.Object{
+		Class: "ArrayClass",
+		ID:    arrayClassID3,
+		Properties: map[string]interface{}{
+			"strings": []string{"a"},
+			"numbers": []float64{1.0},
+		},
+	})
+	assertGetObjectEventually(t, arrayClassID3)
 }

--- a/test/acceptance/objects/auto_schema_test.go
+++ b/test/acceptance/objects/auto_schema_test.go
@@ -54,6 +54,9 @@ func autoSchemaObjects(t *testing.T) {
 							"nationalFormatted":      "020 1234567",
 							"valid":                  true,
 						},
+						"stringArray": []string{"a", "b", "c"},
+						"intArray":    []int{1, 2, 3},
+						"numberArray": []int{11.0, 22.0, 33.0},
 					},
 				}
 			},

